ckan/config/middleware/flask_app.py

```diff
@@ -24,6 +58,81 @@ def make_flask_stack(conf, **app_conf):
         app.config.update(conf)
         app.config.update(app_conf)
 
+    # Do all the Flask-specific stuff before adding other middlewares
+
+    # Secret key needed for flask-debug-toolbar and sessions
+    if not app.config.get('SECRET_KEY'):
+        app.config['SECRET_KEY'] = config.get('beaker.session.secret')
+    if not app.config.get('SECRET_KEY'):
+        raise RuntimeError(u'You must provide a value for the secret key'
+                           ' with the SECRET_KEY config option')
+
+    if debug:
+        app.config['DEBUG_TB_INTERCEPT_REDIRECTS'] = False
+        DebugToolbarExtension(app)
+
+    # Add Jinja2 extensions and filters
+    extensions = [
+        'jinja2.ext.do', 'jinja2.ext.with_',
+        jinja_extensions.SnippetExtension,
+        jinja_extensions.CkanExtend,
+        jinja_extensions.CkanInternationalizationExtension,
+        jinja_extensions.LinkForExtension,
+        jinja_extensions.ResourceExtension,
+        jinja_extensions.UrlForStaticExtension,
+        jinja_extensions.UrlForExtension
+    ]
+    for extension in extensions:
+        app.jinja_env.add_extension(extension)
+    app.jinja_env.filters['empty_and_escape'] = \
+        jinja_extensions.empty_and_escape
+    app.jinja_env.filters['truncate'] = jinja_extensions.truncate
+
+    # Common handlers for all requests
+    @app.before_request
```

Comment:
`app` is only available in the context of the `make_flask_stack` function, so I guess I can't use the decorator at the module level. Is there a way of moving the functions and assigning them directly to `app`? All I could find is `app.before_request_funcs`, but it looks like it not should be messed with directly (the docs say to use the decorator).